packages/hls-parser/src/lib/tags/tagWithAttributesProcessors.ts

```diff
@@ -568,11 +662,23 @@ export class ExtXSessionData extends TagWithAttributesProcessor {
   protected readonly requiredAttributes = new Set([ExtXSessionData.DATA_ID]);
   protected readonly tag = EXT_X_SESSION_DATA;
 
-  protected safeProcess(tagAttributes: Record<string, string>, playlist: ParsedPlaylist): void {
+  protected safeProcess(
+    tagAttributes: Record<string, string>,
+    playlist: ParsedPlaylist,
+    sharedState: SharedState
+  ): void {
+    const uri = tagAttributes[ExtXSessionData.URI];
+    let resolvedUri;
+
+    if (uri) {
```

Comment:
Is there a reason we check for the existence of `uri` here but not in some of the other processors?

Reply:
We check URI existence in different places...
you can search for `if (uri)` in this pr

it depends on whether this attribute is required or not for a particular tag